internal/db/schema/migrations/oss/postgres/99/02_worker_controller_tables.up.sql

```diff
@@ -27,64 +27,82 @@ create trigger controller_update_time_column before update on server_controller
+      ),
   create_time wt_timestamp,
   update_time wt_timestamp,
   version wt_version,
+  type text not null
+    constraint server_worker_type_enm_fkey
+      references server_worker_type_enm (name)
+      on delete restrict
+      on update cascade,
   last_status_time timestamp with time zone,
   constraint last_status_time_not_before_create_time
     check (last_status_time >= create_time),
-  -- This is the calculated address that the worker reports it is reachable on.
-  -- This must be set if the worker has ever received a status update.
-  worker_reported_address wt_network_address
-    constraint worker_reported_address_must_be_set_by_status
-      check (
-          (last_status_time is null and worker_reported_address is null)
-          or
-          (last_status_time is not null and worker_reported_address is not null)
-        ),
-  -- This is the name that the worker reports in it's status updates.
-  worker_reported_name wt_name
-    constraint worker_reported_name_must_be_set_by_status
-      check (
-          (last_status_time is null and worker_reported_name is null)
-          or
-          (last_status_time is not null and worker_reported_name is not null)
-          or
-          (last_status_time is not null and worker_reported_key_id is not null)
-        )
-    constraint worker_reported_name_must_be_lowercase
-      check (lower(trim(worker_reported_name)) = worker_reported_name)
-    constraint worker_reported_name_only_has_printable_characters
-      check (worker_reported_name !~ '[^[:print:]]'),
-  worker_reported_key_id text
-    constraint worker_reported_key_id_must_be_set_by_status
+  key_id text
+    constraint key_id_must_be_set_by_status
       check (
-          (last_status_time is null and worker_reported_key_id is null)
+          (last_status_time is null and key_id is null and type = 'kms')
           or
-          (last_status_time is not null and worker_reported_name is not null)
+          (last_status_time is not null and name is not null and type = 'kms')
           or
-          (last_status_time is not null and worker_reported_key_id is not null)
+          (last_status_time is not null and key_id is not null and type = 'pki')
         ),
   constraint server_worker_scope_id_name_uq
-    unique(scope_id, name),
-  constraint server_worker_scope_id_worker_reported_name_uq
-    unique(scope_id, worker_reported_name)
+    unique(scope_id, name)
 );
 comment on table server_worker  is
   'server_worker is a table where each row represents a Boundary worker.';
 
 create trigger immutable_columns before update on server_worker
-  for each row execute procedure immutable_columns('public_id', 'scope_id', 'create_time');
+  for each row execute procedure immutable_columns('public_id', 'scope_id', 'type', 'create_time');
 
 create trigger default_create_time_column before insert on server_worker
   for each row execute procedure default_create_time();
```

Comment:
I've added a fixme comment.. since it's not critical to write/test this new version trigger before beta.
https://hashicorp.atlassian.net/browse/ICU-4402